etcdutl/etcdutl/check_command.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdutl
+
+import (
+	"encoding/binary"
+	"fmt"
+	"os"
+
+	"go.etcd.io/etcd/pkg/v3/cobrautl"
+	"go.etcd.io/etcd/server/v3/etcdserver/api/snap"
+	"go.etcd.io/etcd/server/v3/storage/backend"
+	"go.etcd.io/etcd/server/v3/storage/datadir"
+	"go.etcd.io/etcd/server/v3/storage/schema"
+	"go.etcd.io/etcd/server/v3/storage/wal"
+
+	"github.com/spf13/cobra"
+	bolt "go.etcd.io/bbolt"
+)
+
+var (
+	checkDataDir           string
+	ConsistentIndexKeyName = []byte("consistent_index")
+)
+
+// NewCheckCommand returns the cobra command for "check".
+func NewCheckCommand() *cobra.Command {
+	cc := &cobra.Command{
+		Use:   "check <subcommand>",
+		Short: "commands for checking properties of the etcd cluster",
+	}
+
+	cc.AddCommand(NewCheckBrokenCommand())
+
+	return cc
+}
+
+// NewCheckBrokenCommand returns the cobra command for "check broken".
+func NewCheckBrokenCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "broken [options]",
+		Short: "Check whether the data file is broken",
+		Run:   newCheckBrokenCommand,
+	}
+
+	cmd.Flags().StringVar(&checkDataDir, "data-dir", "", "Path to the output data directory")
+	cmd.MarkFlagRequired("data-dir")
+
+	return cmd
+}
+
+func newCheckBrokenCommand(cmd *cobra.Command, args []string) {
```

Comment:
Have you seen [Verify](https://github.com/etcd-io/etcd/blob/c1da9dd0f0411293d1c3555494caedabd90926e9/server/verify/verify.go#L49) function? Logic seams to be similar. Maybe it can be reused and command can be renamed to `verify`

Reply:
Thank you Bogdan, 
I'm all for driving this around 'Verify' that is also used in tests to check whether they are living etcd storage in the consistence state and should be expanded with additional checks.